tv_apps/smart-home/js/card_util.js

```diff
@@ -187,19 +188,18 @@
       var nameSpan = document.createElement('span');
       nameSpan.classList.add('name');
       this._localizeCardName(nameSpan, card);
-      cardButton.appendChild(nameSpan);
 
-      return cardButton;
+      var fragment = document.createDocumentFragment();
+      fragment.appendChild(cardButton);
+      fragment.appendChild(nameSpan);
+
+      return fragment;
     },
 
-    updateCardName: function(cardButton, card) {
-      var spans =
-           SharedUtils.nodeListToArray(cardButton.getElementsByTagName('span'));
-      spans.forEach(span => {
-        if (span.classList.contains('name')) {
-          this._localizeCardName(span, card);
-        }
-      });
+    updateCardName: function(nameElement, card) {
+      if (nameElement.classList.contains('name')) {
```

Comment:
I guess the original logic behind the `spans` depends on `SharedUtils.nodeListToArray` so we need to filter that. Since this function is only called by `home.js` now (changes below), we can make sure `nameElement` should be always valid.